File: 100-same-tree/100-same-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        
       if(p == NULL && q == NULL) //if both are null
            return true; 
        
        else if( (p == NULL || q == NULL)) //if one of the node is null
            return false; 
        
         else if ( p->val!=q->val ) //if one of the node is different 
            return false;
                
        return isSameTree(p->left,q->left) && isSameTree(p->right,q->right);
        
    }
};

File: 100-same-tree/100-same-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        # if reaches bottom of tree
        if not p and not q:
            return True
        # if one of the node is empty
        elif not p or not q:
            return False
            # check and calls tree recursively
        else:
            return p.val==q.val and self.isSameTree(p.left,q.left) and self.isSameTree(p.right,q.right)
            
            
            
            
        
       